bufit.go

```diff
@@ -145,6 +145,18 @@ func (b *Buffer) Close() error {
 	return nil
 }
 
+// Len returns the #  of bytes buffered for Readers
+func (b *Buffer) Len() int {
+	return b.buf.Len()
+}
+
+// Discard drops the next n buffered bytes. It returns the actual number of
+// bytes dropped and may return io.EOF if all remaining bytes have been
+// discarded.
+func (b *Buffer) Discard(n int) (int, error) {
+	return b.buf.Discard(n)
```

Comment:
This was fairly tricky, I've implemented a capped version of the buffer though if you want to try it out.